### DISCUSSION TITLE
Custom Menu Options

### DISCUSSION THREAD

**Discussion by agscripter** [josdejong/svelte-jsoneditor#422] [Q&A]
Hello,

Is it possible to customize the menu options?

**Context:**
Right, it provides option to insert: Structure, Object, Array and Value. 
But I would like to have my own options. I'm using this editor for a configurable JSON Form in my system, my JSON has specific structures, so I would like to have an option to Insert specific JSON structures for the form fields, like Inputs, Radio Buttons, Checkbox.

Thank you.

**Comment by josdejong** [comment]
Yes, see option [`onRenderContextMenu`](https://github.com/josdejong/svelte-jsoneditor?tab=readme-ov-file#onrendercontextmenu) in the docs. There is also a usage example in [/src/routes/examples](https://github.com/josdejong/svelte-jsoneditor/tree/main/src/routes/examples).
